Adding reference to OpenSPP-API

kneckinator opened the pull request:
With this reference, it is possible to install the OpenSPP-API apps in OpenSPP.

kneckinator committed:
Adding reference to OpenSPP-API

jeremi commented on odoo/custom/src/repos.yaml:
```diff
@@ -216,7 +229,7 @@ openspp_base:
   target: openspp 15.1.1
   merges:
-    - openspp 15.1.1
+    - openspp module_spp_audit_log
```
Do we want to default on this branch?

kneckinator replied:
Oops. Yes. Thanks for spotting.

jeremi commented on odoo/custom/src/repos.yaml:
```diff
@@ -53,6 +53,19 @@ openspp_registry:
+    openspp: https://github.com/openspp/openspp-api.git
+  target: openspp 15.1.1
+  merges:
+    - openspp spp_dci_api_server
```
Should we do `15.1.1` instead

kneckinator replied:
Valid point. I updated this.

kneckinator replied:
However, there is no `15.1.1` in openspp-api – only `15.0`. How do you suggest that we proceed @jeremi ?

kneckinator committed:
Adjusting repos.yaml

kneckinator committed:
Changing depth level for places where we merge a specific commit
This change is to avoid errors with git-aggregate when it cannot find the commit

kneckinator commented on odoo/custom/dependencies/pip.txt:
```diff
@@ -23,3 +23,11 @@ pydantic
+jwt
+PyLD
+swagger_spec_validator
+cryptography<38
```
Self-comment: This prevents OpenSPP JWT Rest API Authentication from being installed, as it requires pyOpenSSL 22.0.0 (or higher) to be installed.

@reichie020212 @jeremi we need to resolve this by adjusting the code in `openspp_api` to support the latest version of pyOpenSSL.

kneckinator replied:
I will remove the pinning of `cryptography` now.

kneckinator committed:
Reverting the pinning of Cryptography as it interferred with other OpenSPP modules

reichie020212 committed:
changed jwt to pyjwt

kneckinator commented:
PR is stale and also no longer relevant. Closing it.
